src/oop/special_class_method/hard/q2/MainTest.java

```diff
@@ -0,0 +1,249 @@
+package oop.special_class_method.hard.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "Glasses";
+    private final String doubleFieldName = "size";
+    private final String stringFieldName = "type";
+    private final String setStringMethodName = "setType";
+    private final String getStringMethodName = "getType";
+    private final String getDoubleMethodName = "getSize";
+    private final String setDoubleMethodName = "setSize";
+    private final String testClassName = "TestClass";
+    private final double outputSize = 7.5;
+    private final String outputType = "Long";
+    private ObjectTest testClass;
+    private ObjectTest outputClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q2." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.special_class_method.hard.q2." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+    }
+
+    @Test
+    public void correctTestClass() throws Throwable {
```

Comment:
I would add tags to this to indicate this test is dependent on other tests.

Reply:
I mean, it isn't really? In your example you didn't tag the constructor or the toString with the "dependent1" tag and this test class only uses the constructor and toString so there's no point tagging dependency1 without at least one dependent1